### PR TITLE
[Octo-Bot] Adding new healthcare-registration-bot

### PR DESCRIPTION
**Pull request by joelczk**

### CODE REVIEW

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+You need to have Git to clone this repositories onto your desired directory.
+
+Installing git:
+```console
+joelczk@OctoBot:~$ sudo apt update && sudo apt install git
+joelczk@OctoBot:~$ git clone https://github.com/nus-ncl/OctoBot.git
+```
+
+### Using Docker
+To use docker container to run the bot, you need to install `docker`. On Ubuntu, `docker` can be installed with the following commands:
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/README.md:23]: Use capital letter for name.
```suggestion
To use Docker container to run the bot, you need to install `Docker`. On Ubuntu, `Docker` can be installed with the following commands:
```

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+```console
+joelczk@OctoBot:~$ sudo apt-get update
+joelczk@OctoBot:~$ sudo apt-get install docker-ce docker-ce-cli containerd.io
+joelczk@OctoBot:~$ sudo docker run hello-world
+```
+
+For other distributions, visit [here](https://docs.docker.com/engine/install/). It is not recommended to use Docker using `Windows` OS.
+
+You can then build the docker image using the provided script like this:
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/README.md:32]:
```suggestion
You can then build the Docker image using the provided script like this:
```

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+Finally you can run the bot with `docker run`. Here is an example:
+```console
+joelczk@OctoBot/Octo-Bot/<bot-name>:~$ docker run healthcare-registration-bot:latest [-u hostname]
+```
+
+## Developing the bot
+
+Create a local build of the docker image:
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/README.md:45]:
```suggestion
Create a local build of the Docker image:
```

Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/README.md:45]:
```suggestion
Create a local build of the Docker image:
```

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+Check the local image:
+
+```console
+joelczk@OctoBot:~/OctoBot/Octo-Bot/<bot-name>$ docker images
+REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
+healthcare-registration-bot    latest              f112835c8a07        1 days ago          1.1GB
+python                         3.8-alpine3.10      02d2bb146b3b        2 weeks ago         918MB
+```
+Run the docker image:
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/README.md:60]:
```suggestion
Run the Docker image:
```

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+Here is an example format of `patient.csv` file:
+```csv
+<username>, <password>,<diagnosis code>
+```
+
+Here is an example format of `patient.csv` file:
+```csv
+<username>, <dob(yyyy-mm-dd)>, <first name>, <Last name>, <address>, <email>, <postal code>, <contact number>, <password>, <Field to be automated(height/weight/temp/bp)>, <data of field to be automated>
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/README.md:114]:
```suggestion
<username>, <dob(yyyy-mm-dd)>, <first name>, <last name>, <address>, <email>, <postal code>, <contact number>, <password>, <Field to be automated(height/weight/temp/bp)>, <data of field to be automated>
```

File: Octo-Bot/healthcare-registration-bot/build.sh [hunk 6]
```diff
@@ -0,0 +1,3 @@
+chmod a+x *.sh &&
+
+sudo docker build . -t healthcare-registration-bot:latest
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/build.sh:3]: Add a newline at the last line.
```suggestion
sudo docker build . -t healthcare-registration-bot:latest

```

File: Octo-Bot/healthcare-registration-bot/main.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        help='URL of the website that the bot is using', default = "https://10.10.0.112/")
+    args = parser.parse_args()
+    url = args.u
+    urlPrintLine = "Preparing to access from " + str(url) + "..."
+    print(urlPrintLine)
+    print("Registration bot starting up in 5...")
+    print("Registration bot starting up in 4...")
+    print("Registration bot starting up in 3...")
+    print("Registration bot starting up in 2...")
+    print("Registration bot starting up in 1...")
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/main.py:65]: I think it is not really important, except there is any checking or counter need to be satisfied.

File: Octo-Bot/healthcare-registration-bot/main.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    main(url)
+    print("Registration bot completed and closing down in 5...")
+    print("Registration bot completed and closing down in 4...")
+    print("Registration bot completed and closing down in 3...")
+    print("Registration bot completed and closing down in 2...")
+    print("Registration bot completed and closing down in 1...")
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/main.py:71]: I think it is not really important, except there is any checking or counter need to be satisfied.

File: Octo-Bot/healthcare-registration-bot/test.sh [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+fi
+pip3 install selenium >/dev/null
+
+echo "All dependencies have been fulfilled!"
+
+# Run all unit tests
+python3 test/test.py
+
+echo
+echo "Please check if tests succeeded!"
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test.sh:15]:
```suggestion
echo "Please check if tests succeeded!"

```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+    try:
+        driver.execute(Command.STATUS)
+        return driverAlive
+    except:
+        return driverDead
+        
+def registerPatientAccount(url, driver):
+
+    '''
+    Main Logic used for registering a patient account
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:141]:
```suggestion
    Main logic used for registering a patient account
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+
+    Arguments:
+        url(str): Target URL
+        driver(obj): firefox webdriver instance in python
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:145]:
```suggestion
        driver(obj): Firefox webdriver instance in python 
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+    driver.find_element_by_id('BodyContent_inputContactNumber').send_keys(contactNumber)
+    driver.find_element_by_id('BodyContent_inputPassword').send_keys(password)
+    driver.find_element_by_id('BodyContent_inputPasswordConfirm').send_keys(password)
+    driver.find_element_by_id('BodyContent_inputCountryofBirth').send_keys('Afghanistan')
+    driver.find_element_by_id('BodyContent_inputNationality').send_keys('Afghan')
+    driver.find_element_by_id('BodyContent_buttonRegister').click()
+
+def assignTherapist(url, driver):
+    '''
+    Main Logic behind assigning therapist for the patient
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:179]:
```suggestion
    Main logic behind assigning therapist for the patient
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+
+    Arguments:
+        url(str): Target URL
+        driver(obj): firefox webdriver instance in python
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:183]:
```suggestion
        driver(obj): Firefox webdriver instance in python 
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+    logoutScript = "__doPostBack('ctl00$ctl13','')"
+    driver.execute_script(logoutScript)
+    driver.close()
+    driver.quit()
+    print(" -Therapist assigned to patient")
+
+def createNewRecord(url, driver):
+
+    '''
+    Main Logic behind creating new records for patient
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:211]:
```suggestion
    Main logic behind creating new records for patient
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 15]
```diff
@@ -0,0 +1,360 @@
...
+
+    Arguments:
+        url(str): Target URL
+        driver(obj): firefox webdriver instance in python
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:215]:
```suggestion
        driver(obj): Firefox webdriver instance in python 
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 16]
```diff
@@ -0,0 +1,360 @@
...
+    logoutScript = "__doPostBack('ctl00$ctl13','')"
+    driver.execute_script(logoutScript)
+    driver.close()
+    driver.quit()
+    print(" -Records created for new patient")
+
+def requestPermissions(url, driver):
+
+    '''
+    Main Logic behind requesting permission for therapist to access patient records
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:247]:
```suggestion
    Main logic behind requesting permission for therapist to access patient records
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 17]
```diff
@@ -0,0 +1,360 @@
...
+
+    Arguments:
+        url(str): Target URL
+        driver(obj): firefox webdriver instance in python
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:251]:
```suggestion
        driver(obj): Firefox webdriver instance in python 
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 18]
```diff
@@ -0,0 +1,360 @@
...
+        print(" -Therapist request for permission")
+    except:
+        driver.close()
+        driver.quit()
+        print("Permission already obtained")
+
+def approvePermissions(url, driver):
+
+    '''
+    Main Logic behind approving permissions for therapist to access patient records
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:293]:
```suggestion
    Main logic behind approving permissions for therapist to access patient records
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 19]
```diff
@@ -0,0 +1,360 @@
...
+
+    Arguments:
+        url(str): Target URL
+        driver(obj): firefox webdriver instance in python
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:297]:
```suggestion
        driver(obj): Firefox webdriver instance in python 
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 20]
```diff
@@ -0,0 +1,360 @@
...
+    logoutScript = "__doPostBack('ctl00$ctl13','')"
+    driver.execute_script(logoutScript)
+    driver.close()
+    driver.quit()
+    print(" -Patient approved permission for therapist")
+
+def addDiagnosis(url, driver):
+
+    '''
+    Main Logic behind addition of diagnosis for patient
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:315]:
```suggestion
    Main logic behind addition of diagnosis for patient
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 21]
```diff
@@ -0,0 +1,360 @@
...
+
+    Arguments:
+        url(str): Target URL
+        driver(obj): firefox webdriver instance in python
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:319]:
```suggestion
        driver(obj): Firefox webdriver instance in python 
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 22]
```diff
@@ -0,0 +1,360 @@
...
+    logoutScript = "__doPostBack('ctl00$ctl13','')"
+    driver.execute_script(logoutScript)
+    driver.close()
+    driver.quit()
+    print(" -Therapist has created a new diagnosis")
+
+def viewDiagnosis(url, driver):
+
+    '''
+    Main Logic behind view diagnosis for patients
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:347]:
```suggestion
    Main logic behind view diagnosis for patients
```

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py [hunk 23]
```diff
@@ -0,0 +1,360 @@
...
+    Arguments:
+        url(str): Target URL
+        driver(obj): firefox webdriver instance in python
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/BotActions.py:350]:
```suggestion
        driver(obj): Firefox webdriver instance in python 
```

File: Octo-Bot/healthcare-registration-bot/test/main.py [hunk 24]
```diff
@@ -0,0 +1,71 @@
...
+        help='URL of the website that the bot is using', default = "https://10.10.0.112/")
+    args = parser.parse_args()
+    url = args.u
+    urlPrintLine = "Preparing to access from " + str(url) + "..."
+    print(urlPrintLine)
+    print("Registration bot starting up in 5...")
+    print("Registration bot starting up in 4...")
+    print("Registration bot starting up in 3...")
+    print("Registration bot starting up in 2...")
+    print("Registration bot starting up in 1...")
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/main.py:65]: I think it is not really important, except there is any checking or counter need to be satisfied.

File: Octo-Bot/healthcare-registration-bot/test/main.py [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+    main(url)
+    print("Registration bot completed and closing down in 5...")
+    print("Registration bot completed and closing down in 4...")
+    print("Registration bot completed and closing down in 3...")
+    print("Registration bot completed and closing down in 2...")
+    print("Registration bot completed and closing down in 1...")
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/main.py:71]: I think it is not really important, except there is any checking or counter need to be satisfied.

File: Octo-Bot/healthcare-registration-bot/test/test.py [hunk 26]
```diff
@@ -0,0 +1,86 @@
...
+    ''' 
+    def setUp(self):
+        warnings.simplefilter('ignore',category = DeprecationWarning)
+        warnings.simplefilter('ignore', category = ResourceWarning)
+        text_trap = io.StringIO()
+        sys.stdout = text_trap
+
+    def testGetAdminCredentials(self):
+        adminCredentials = getAdminCredentials()
+        adminCredentialsData = ["S0000001A", "DrasticP@ssw0rdPlayer"]
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/test.py:32]: Can we hide this password? Ex. use our own defined admin.csv

File: Octo-Bot/healthcare-registration-bot/test/test.py [hunk 27]
```diff
@@ -0,0 +1,86 @@
...
+        self.assertEqual(adminCredentials, adminCredentialsData)
+    
+    def testGetTherapistCredentials(self):
+        therapistCredentials = getTherapistCredentials()
+        therapistCredentialsData = ["F0058132Q", "LightEnterP@ssw0rd", "Z9989"]
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/test.py:37]: Can we hide this password? Ex. use our own defined therapist.csv

File: Octo-Bot/healthcare-registration-bot/test/test.py [hunk 28]
```diff
@@ -0,0 +1,86 @@
...
+        self.assertEqual(therapistCredentials, therapistCredentialsData)
+    
+    def testGetPatientCredentials(self):
+        patientCredentials = getPatientCredentials()
+        patientCredentialsData = ["S1757946A", "1995-12-12", "John", "Tan", "Block 5002 Ang Mo Kio Avenue 5 04-05 TECHplace II Singapore 569871  Singapore", "john@email.com", "569871", "12345678", "veryEasyP@ssw0rd", "Weight", "20"]
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/test/test.py:42]: Can we hide this password? Ex. use our own defined patient.csv

File: Octo-Bot/healthcare-registration-bot/README.md (hunk 1 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md (hunk 2 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md (hunk 3 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved

Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/main.py (hunk 24 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/main.py]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/test.py (hunk 26 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/test.py]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/test.py (hunk 27 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/test.py]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/main.py (hunk 25 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/main.py]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md (hunk 4 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md (hunk 5 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md:114]: Resolved

File: Octo-Bot/healthcare-registration-bot/build.sh (hunk 6 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/build.sh:3]: Resolved!

File: Octo-Bot/healthcare-registration-bot/main.py (hunk 8 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/main.py]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 10 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:141]: Resolved!

File: Octo-Bot/healthcare-registration-bot/main.py (hunk 7 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/main.py]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 12 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:179]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 11 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:145]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 13 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:183]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 14 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:211]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 16 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:247]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 18 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:293]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 20 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:315]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 22 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:347]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 23 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:350]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/test.py (hunk 28 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/test.py:42]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 21 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:319]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 19 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:297]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 17 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:251]: Resolved!

File: Octo-Bot/healthcare-registration-bot/test/BotActions.py (hunk 15 above)
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/test/BotActions.py:215]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 29]
```diff
@@ -20,53 +20,53 @@ joelczk@OctoBot:~$ git clone https://github.com/nus-ncl/OctoBot.git
...
 
 ### Using Docker
-To use docker container to run the bot, you need to install `docker`. On Ubuntu, `docker` can be installed with the following commands:
+To use Docker container to run the bot, you need to install `Docker`. On Ubuntu, `Docker` can be installed with the following commands:
 ```console
 joelczk@OctoBot:~$ sudo apt-get update
-joelczk@OctoBot:~$ sudo apt-get install docker-ce docker-ce-cli containerd.io
-joelczk@OctoBot:~$ sudo docker run hello-world
+joelczk@OctoBot:~$ sudo apt-get install Docker-ce Docker-ce-cli containerd.io
+joelczk@OctoBot:~$ sudo Docker run hello-world
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 30]
```diff
@@ -20,53 +20,53 @@ joelczk@OctoBot:~$ git clone https://github.com/nus-ncl/OctoBot.git
...
 ```
 
-For other distributions, visit [here](https://docs.docker.com/engine/install/). It is not recommended to use Docker using `Windows` OS.
+For other distributions, visit [here](https://docs.Docker.com/engine/install/). It is not recommended to use Docker using `Windows` OS.
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 31]
```diff
@@ -20,53 +20,53 @@ joelczk@OctoBot:~$ git clone https://github.com/nus-ncl/OctoBot.git
...
 
-You can then build the docker image using the provided script like this:
+You can then build the Docker image using the provided script like this:
 ```console
 joelczk@OctoBot/Octo-Bot/<bot-name>:~$chmod 755 build.sh
 joelczk@OctoBot/Octo-Bot/<bot-name>:~$./build.sh 
 ```
 
-Finally you can run the bot with `docker run`. Here is an example:
+Finally you can run the bot with `Docker run`. Here is an example:
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 32]
```diff
@@ -20,53 +20,53 @@ joelczk@OctoBot:~$ git clone https://github.com/nus-ncl/OctoBot.git
...
 ```console
-joelczk@OctoBot/Octo-Bot/<bot-name>:~$ docker run healthcare-registration-bot:latest [-u hostname]
+joelczk@OctoBot/Octo-Bot/<bot-name>:~$ Docker run healthcare-registration-bot:latest [-u hostname]
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 33]
```diff
@@ -20,53 +20,53 @@ joelczk@OctoBot:~$ git clone https://github.com/nus-ncl/OctoBot.git
...
 
 ## Developing the bot
 
-Create a local build of the docker image:
+Create a local build of the Docker image:
 
 ```console
 joelczk@OctoBot:~$ cd OctoBot/Octo-Bot/<bot-name>
-joelczk@OctoBot/Octo-Bot/<bot-name>:~/OctoBot$ docker build . -t healthcare-registration-bot:latest
+joelczk@OctoBot/Octo-Bot/<bot-name>:~/OctoBot$ Docker build . -t healthcare-registration-bot:latest
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 34]
```diff
@@ -20,53 +20,53 @@ joelczk@OctoBot:~$ git clone https://github.com/nus-ncl/OctoBot.git
...
 ```
 
 Check the local image:
 
 ```console
-joelczk@OctoBot:~/OctoBot/Octo-Bot/<bot-name>$ docker images
+joelczk@OctoBot:~/OctoBot/Octo-Bot/<bot-name>$ Docker images
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 35]
```diff
@@ -20,53 +20,53 @@ joelczk@OctoBot:~$ git clone https://github.com/nus-ncl/OctoBot.git
...
 REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
 healthcare-registration-bot    latest              f112835c8a07        1 days ago          1.1GB
 python                         3.8-alpine3.10      02d2bb146b3b        2 weeks ago         918MB
 ```
-Run the docker image:
+Run the Docker image:
 
 ```console
-joelczk@OctoBot/Octo-Bot/<bot-name>:~/OctoBot$ docker run healthcare-registration-bot -u <url>
+joelczk@OctoBot/Octo-Bot/<bot-name>:~/OctoBot$ Docker run healthcare-registration-bot -u <url>
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 36]
```diff
@@ -78,7 +78,7 @@ CONTAINER ID        IMAGE                                COMMAND
 ## Usage
 Here is the help page for the healthcare-registration-bot program:
 ```console
-joelczk@OctoBot:~/OctoBot/Octo-Bot/<bot-name>$ docker run healthcare-registration-bot -h
+joelczk@OctoBot:~/OctoBot/Octo-Bot/<bot-name>$ Docker run healthcare-registration-bot -h
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 37]
```diff
@@ -93,7 +93,7 @@ optional arguments:
 ## Specify hostname for bot to crawl
 This specifies the hostname that the bot is crawling. The default hostname is `https://10.10.0.112/` if it is not specified.
 ```console
-joelczk@OctoBot:~/OctoBot/Octo-Bot/<bot-name>$ docker run healthcare-registration-bot -u https://10.10.0.112/
+joelczk@OctoBot:~/OctoBot/Octo-Bot/<bot-name>$ Docker run healthcare-registration-bot -u https://10.10.0.112/
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved!

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 38]
```diff
@@ -113,3 +113,18 @@ Here is an example format of `patient.csv` file:
...
+```console
+joelczk@OctoBot/Octo-Bot/<bot-name>/test:~/OctoBot$ python3 TestUI.py
+```
+
+### Running tests without Firefox interface UI
+```console
+joelczk@OctoBot/Octo-Bot/<bot-name>:~/OctoBot$ TestNoUI.py
+```
+
+By default, `test.sh` file runs both tests with and without Firefox interface UI
```
Comment (ariscahyadi) [inline Octo-Bot/healthcare-registration-bot/README.md:130]:
```suggestion
By default, `test.sh` file runs both tests with and without Firefox interface UI (it is required desktop terminal)
```

Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved.

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 39]
```diff
@@ -113,3 +113,18 @@ Here is an example format of `patient.csv` file:
 ```csv
 <username>, <dob(yyyy-mm-dd)>, <first name>, <last name>, <address>, <email>, <postal code>, <contact number>, <password>, <Field to be automated(height/weight/temp/bp)>, <data of field to be automated>
 ```
+
+## Tests
+There are 2 types of tests that can be run for `healthcare-registration-bot`, namely tests with Firefox interface UI and tests without Firefox interface UI
+
+### Running tests with Firefox interface UI
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved.

File: Octo-Bot/healthcare-registration-bot/README.md [hunk 40]
```diff
@@ -113,3 +113,18 @@ Here is an example format of `patient.csv` file:
...
+```console
+joelczk@OctoBot/Octo-Bot/<bot-name>/test:~/OctoBot$ python3 TestUI.py
+```
+
+### Running tests without Firefox interface UI
+```console
+joelczk@OctoBot/Octo-Bot/<bot-name>:~/OctoBot$ TestNoUI.py
```
[the remark this answers is not in the document]
Comment (joelczk) [reply Octo-Bot/healthcare-registration-bot/README.md]: Resolved.

**Pull request merged**